packages/core/src/Select/Select.stories.tsx

```diff
@@ -0,0 +1,183 @@
+import { useState } from "react";
+import { Decorator, Meta, StoryObj } from "@storybook/react";
+
+import { HvSelect, HvSelectProps } from "./Select";
+import { HvButton } from "../Button";
+import { HvOption } from "./Option";
+import { HvOptionGroup } from "./OptionGroup";
+
+const decorator: Decorator = (Story) => (
+  <div className="w-[300px] min-h-[300px]">{Story()}</div>
+);
+
+export default {
+  title: "Components/Select",
+  component: HvSelect,
+  // @ts-expect-error https://github.com/storybookjs/storybook/issues/20782
+  subcomponents: { HvOption, HvOptionGroup },
+} satisfies Meta<typeof HvSelect>;
+
+const options = [
+  { value: "ar", label: "Argentina", flag: "🇦🇷" },
+  { value: "bg", label: "Belgium", flag: "🇧🇪" },
+  { value: "pt", label: "Portugal", flag: "🇵🇹" },
+  { value: "pl", label: "Poland", flag: "🇵🇱" },
+  { value: "sp", label: "Spain", flag: "🇪🇸" },
+  { value: "us", label: "United States", flag: "🇺🇸" },
+];
+
+export const Main: StoryObj<HvSelectProps<{}, false>> = {
+  args: {
+    multiple: false,
+  },
+  argTypes: {},
+  decorators: [decorator],
+  render: (args) => {
+    return (
+      <HvSelect
+        required
+        name="country"
+        label="Country"
+        description="Select your favorite country"
+        placeholder="Select country"
+        onChange={(evt, val) => console.log(val)}
+        {...args}
+      >
+        <HvOptionGroup label="America">
+          <HvOption value="ar">Argentina</HvOption>
+          <HvOption value="us">United States</HvOption>
+        </HvOptionGroup>
+        <HvOptionGroup label="Europe">
+          <HvOption value="bg">Belgium</HvOption>
+          <HvOption value="pt">Portugal</HvOption>
+          <HvOption value="pl">Poland</HvOption>
+          <HvOption value="sp">Spain</HvOption>
+        </HvOptionGroup>
+      </HvSelect>
+    );
+  },
+};
+
+export const Variants: StoryObj<HvSelectProps<{}, false>> = {
```

Comment:
For some reason, I can't close the dropdown by clicking outside in this sample. I can only close it by clicking on the dropdown again.

Reply:
I aligned the behaviour with the native Select. If there are no items/`children`, it shouldn't open
But I've also improved the sample to have an item to better display the variants